src/components/SpecialResolution/SpecialResolutionForm.vue

```diff
@@ -0,0 +1,463 @@
+  }
+
+  /** The minimum date that can be entered (can't be earlier than incorporation date ). */
+  get resolutionDateMinStr (): string {
+    return this.dateToYyyyMmDd(new Date(this.getBusinessFoundingDate))
+  }
+
+  /** The minimum date that can be entered (can't be earlier than incorporation date ). */
+  get resolutionDateMin (): Date {
+    return this.getBusinessFoundingDate
+  }
+  /** The maximum date that can be entered (today). */
+  get resolutionDateMax (): Date {
+    const date = new Date()
+    return date
+  }
+
+  get resolutionTextRules (): Array<Function> {
+    return [
+      v => (v && v.trim().length > 0) || 'Resolution text is required',
+      v => (v && v.length <= this.MAX_RESOLUTION_TEXT_LENGTH) || 'Maximum characters exceeded',
+      v => (v && /^([\w\s$&+,:;=?@#|'<>.^*()%!-\\"]*)$/g.test(v)) || 'Invalid characters'
+    ]
+  }
+
+  /** The maximum date that can be entered (today). */
+  get resolutionDateMaxStr (): string {
+    const date = new Date()
+    return this.dateToYyyyMmDd(date)
+  }
+
+  /** Validations rules for resolution date field. */
+  get resolutionDateRules (): Array<Function> {
+    return [
+      (v: string) => !!v || 'Resolution date is required',
+      (v: string) =>
+
+        this.isBetweenDates(this.resolutionDateMinStr,
+          this.resolutionDateMaxStr,
+          v) ||
+        `Date should be between ${this.dateToPacificDate(this.resolutionDateMin, true)} and
+         ${this.dateToPacificDate(this.resolutionDateMax, true)}`
+    ]
+  }
+
+  /**
+   * True if date is >= the minimum (ie, today) and <= the maximum (ie, the 10th day).
+   * This is used for Vue form validation (in Date Rules above).
+   */
+  // private isBetweenDates (v: string): boolean {
+  isBetweenDates (minDate: string, maxDate: string, dateStrToValidate: string): boolean {
+    if (!dateStrToValidate) { return true }
+    return (new Date(dateStrToValidate) >= new Date(minDate)) && (new Date(dateStrToValidate) <= new Date(maxDate))
+  }
+  /** Called to add a new date. */
+  async onResolutionDateSync (val: string): Promise<void> {
+    await this.$nextTick()
+    this.resolutionDateText = val
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      resolutionDate: val
+    })
+  }
+  /** called to add new resolutionDateText. */
+  protected onResolutionTextChanged (val: string) {
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      resolutionText: val
+    })
+  }
+
+  /** Validation rule for name */
+  nameRules (label, isRequired = true): Array<Function> {
+    return [
+      v => isRequired ? (!!v?.trim() || `${label} is required`) : true,
+      v => (v?.length <= 30) || 'Cannot exceed 30 characters' // maximum character count
+    ]
+  }
+
+  /** Validation rule for individual name fields */
+  firstNameRules = this.nameRules('First Name');
+  middleNameRules = this.nameRules('Middle Name', false);
+  lastNameRules = this.nameRules('Last Name');
```

Comment:
Don't do this. Declare all class variables at the top with the others. Also should not need semi-colons.

Reply:
Maybe these rules can be readonly also?